### PR TITLE
fix: Inject hostInit as entry if there are no html entry files

### PR DESCRIPTION
**Pull request by RobinBomkamp**
fix:
#189

### CODE REVIEW

File: src/plugins/pluginAddEntry.ts [hunk 1]
```diff
@@ -127,7 +141,7 @@ const addEntry = ({
         }
       },
       transform(code, id) {
-        if (inject === 'entry' && entryFiles.some((file) => id.endsWith(file))) {
+        if (injectEntry() && entryFiles.some((file) => id.endsWith(file))) {
```
Comment (gioboa) [inline src/plugins/pluginAddEntry.ts:143]: Thanks for your help @RobinBomkamp 
Here we are changing the behaviour for the non SSR scenario.
Am I missing something?

Comment (RobinBomkamp) [reply src/plugins/pluginAddEntry.ts:143]: It should not change the non SSR scenario. 

There are 3 times the add-entry-plugin will be called two time with 'inject:entry' and one time with 'inject:html'. 

If the value is 'entry' there is everything normal. Only if the value is 'html' then there will be a different behavior if your build is an ssr build.

Comment (gioboa) [reply src/plugins/pluginAddEntry.ts:143]:
```
function injectEntry() {
  return inject === 'entry' || _ssrBuild; <-- so we can remove this I guess
}
```

Comment (RobinBomkamp) [reply src/plugins/pluginAddEntry.ts:144]: No, that is the difference needed to fix the issue. 

Html should be injected like an 'entry' in ssr scenarios.

**Comment by RobinBomkamp** [comment]
Is it correct that the tests fail?

I couldn't get it running on my local machine.

File: src/plugins/pluginAddEntry.ts (hunk 1 above)
Comment (RobinBomkamp) [reply src/plugins/pluginAddEntry.ts:143]: It seems like there is a bug in my ssr detection. I will fix it.

Comment (RobinBomkamp) [reply src/plugins/pluginAddEntry.ts:143]: The merge request is now ready.

I have removed the ssr detection. The underlying problem was that there is no html file in a ssr build. However the htmlFilePath value was set to the first entry file. But there was no check if the first entry file is an html file.

More over if there is no html file injections with the type html will be injected like an entry.

**Review by gioboa — APPROVED** [review]
Great :+1: Thanks @RobinBomkamp 
I'm really happy to merge your first PR in this project, kudos :partying_face:

**Comment by koga73** [comment]
Is there a way to instruct federation to import into entry instead of html? For my use case 3rd party applications consume my js and I want it bundled instead of having to instruct them to add a separate import for hostInit.

**Pull request merged**